### DISCUSSION TITLE
clickable TreeView items

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#102] [General]
Hello,
how can I implement a TreeView, where I can have multiple root items, and multiple nodes.
The nodes should be clickable, and can have multiple nodes self.
If it possible, the TreeView should have lines to see the Level (but this is a pro, and must not have,
but the Levels should be intend, to see, in which Level the node item is)
As example:

```
root1
  node1
    node11
  node12
    node121
root2
  node2
```
...
Thanks for helping
paule32

**Comment by magiblot** [comment]
Hi Jens!

The TOutline class provides this behaviour. Here you can find an example application that uses TOutline:

https://github.com/magiblot/tvision/blob/9c9608e03b6df011b811f284b4c73459a7246c2d/examples/tvdir/tvdir.cpp

And here you can find a more complex example, from the Turbo text editor:

https://github.com/magiblot/turbo/blob/ba322d6c123bb632c5f2bd6e0c3792db0c116e1c/source/turbo/doctree.h
https://github.com/magiblot/turbo/blob/ba322d6c123bb632c5f2bd6e0c3792db0c116e1c/source/turbo/doctree.cc
